jupyterhub_singleuser_profiles/profiles.py

```diff
@@ -296,11 +298,34 @@ def apply_gpu_config(self, gpu_mode, gpu_count, pod):
     return pod
 
   @classmethod
-  def apply_pod_profile(self, spawner, pod, profile):
+  def generate_volume_path(self, mountPath, default_mount_path):
+    if (mountPath):
+      if (os.path.isabs(mountPath)):
+        return mountPath
+      return os.path.join(default_mount_path, mountPath)
+    return default_mount_path
+    
+
+  @classmethod
+  def apply_pod_profile(self, spawner, pod, profile, default_mount_path):
     api_client = kubernetes.client.ApiClient()
 
     pod.metadata.labels['jupyterhub.opendatahub.io/user'] = escape(spawner.user.name)
 
+    profile_volumes = profile.get('volumes')
+
+    if profile_volumes:
+      for volume in profile_volumes:
+        volume_name = volume['name']
+        read_only = volume['persistentVolumeClaim'].get('readOnly')
+        if read_only == None:
```

Comment:
I don't think you need to check this - None behaves basically like False in Python